src/aerialmap_display.cpp

```diff
@@ -468,51 +466,81 @@ void AerialMapDisplay::assembleScene()
   tileCache_.purge({ tileId, blocks_ });
 }
 
+// rotation matrix for rotating the current frame to ENU
+Ogre::Matrix3 rotateToENU(int convention)
+{
+  if (convention == FRAME_CONVENTION_XYZ_ENU)
+  {
+    return Ogre::Matrix3::IDENTITY;
+  }
+  else if (convention == FRAME_CONVENTION_XYZ_NED)
+  {
+    // XYZ->NED will cause the map to appear reversed when viewed from above (from +z).
+    // clang-format off
+    Ogre::Matrix3 const xyz_R_ned(0, 1, 0,
+                                  1, 0, 0,
+                                  0, 0, -1);
+    // clang-format on
+    return xyz_R_ned.Transpose();
+  }
+  else if (convention == FRAME_CONVENTION_XYZ_NWU)
+  {
+    // clang-format off
+    Ogre::Matrix3 const xyz_R_nwu(0, -1, 0,
+                                  1, 0, 0,
+                                  0, 0, 1);
+    // clang-format on
+    return xyz_R_nwu.Transpose();
+  }
+  else
+  {
+    assert(false);
+  }
+}
+
 void AerialMapDisplay::transformAerialMap()
 {
   if (!isEnabled())
   {
     return;
   }
 
-  // the parent frame of this scene node
-  static std::string const mapFrame = "map";
+  static std::string const frameMap = "map";
+  std::string const frameNavSatFix = ref_fix_.header.frame_id;
 
-  // the robot's frame
-  std::string const frame = ref_fix_.header.frame_id;
+  Ogre::Quaternion orientation, orientationOffset, orientationUnused;
+  Ogre::Vector3 position, positionUnused;
 
-  // note: the orientation is not used on purpose
-  Ogre::Quaternion orientation;
-  Ogre::Vector3 position;
-
-  // can mapFrame be used as a fixed frame?
-  std::string errMsg;
-  if (context_->getFrameManager()->frameHasProblems(mapFrame, ros::Time{}, errMsg) ||
-      context_->getFrameManager()->transformHasProblems(mapFrame, ros::Time{}, errMsg))
+  if (!context_->getFrameManager()->getTransform(frameMap, ref_fix_.header.stamp, positionUnused, orientation))
   {
-    context_->getFrameManager()->setFixedFrame(lastFixedFrame_);
-    setStatus(StatusProperty::Error, "Transform", QString::fromStdString(errMsg));
+    // display error
+    std::string error;
+    if (context_->getFrameManager()->transformHasProblems(frameMap, ref_fix_.header.stamp, error))
```

Comment:
There are a lot of handlings of this in here, and they slightly differ. could this be moved out of here? resp. handled in an own function beforehand? If any of these fail, the whole functions fails, right? it clutters what is actually happening. And since the conversions are not trivial, it'd be nice not to be distracted by all those transform-error-checks in here.

Reply:
Improved quite a bit. I'd still like to separate concerns of looking up frames and calculus on them. But its up to you :)